src/include/brain/brain_defs.h

```diff
@@ -70,6 +61,10 @@ enum class ExecutionOperatingUnitType : uint32_t {
    * IDX_SCAN
    * num_rows: size of index
    * cardinality: size of scan
+   *
+   * Special case: in the case of an index join, it is also recorded as an IDX_SCAN where:
+   * num_rows: number of rows output by the outer loop
+   * cardinality: N/A
```

Comment:
@malin1993ml @17zhangw Can you confirm this N/A part? Or should cardinality be the same as NUM_ROWS?

Reply:
`num_rows` and `cardinality` would actually be the same as a regular index scan. That is, the size of the index and the size of the scan per `idx_scan` invocation. If calculating a per-invocation value is not convenient, you can also just use the value of the last invocation as an approximation.

Reply:
I do not understand what num_loops would be recorded as then (the number of times the outer loop is accessed).

Reply:
That said, I have pushed my changes.

Reply:
I took a look at the changes to the `IndexJoinTranslator`. So the existing `num_scans_index_` essentially tracks how many tuples are retrieved from the index. So this would be recorded as the `cardinality` (which is what is done now).

For `num_loops`, it would essentially be tracking the number of times `@indexIteratorScanKey(&index_iter)` is invoked (which equals # tuples emit by the index join plan's child and equals # times the index is looked up on).